admin/bin/check-ps-report-for-corrupt-data.js

```diff
@@ -0,0 +1,380 @@
+#!/usr/bin/env node
+'use strict'
+const fs = require('fs')
+const csv = require('fast-csv')
+const startTime = Date.now()
+const R = require('ramda')
+const RA = require('ramda-adjunct')
+
+// const timeRegex =  /^\d{1,2}:\d{2}:\d{2} (AM|PM)$/i // Not quite right - PM instead pm and single digit hours
+const timeRegex = /^\d{2}:\d{2}:\d{2} (AM|PM)$/ // Correct
+
+// Check for corrupt data in the PS report
+function help () {
+  console.log(`usage: ${process.argv[0]} <ps-file-to-inspect>`)
+}
+
+function report (prop, row, val) {
+  // console.log()
+  console.log(`${prop} field not matched for ${row['AttemptId']} val was '${val}'`)
+}
+
+function reportMissing (row, prop) {
+  process.stderr.write(`${prop} field missing for ${row['AttemptId']}\n`)
+}
+
+function reportProgress (count) {
+  // process.stdout.write('.')
+}
+
+function regexTest (row, prop, regex) {
+  if (row.hasOwnProperty(prop)) {
+    const val = row[prop]
+    if (val.length && !regex.test(val)) {
+      report(prop, row, val)
+    }
+  } else {
+    console.log(`Missing property: ${row['checkCode']} property: ${prop}`)
+  }
+}
+
+const checkID = function (row, prop) {
+  const regex = /^\d{1,2} x \d{1,2}$/
+  regexTest(row, prop, regex)
+}
+
+const checkResponse = function (row, prop) {
+  const regex = /^\d{0,5}$/
+  regexTest(row, prop, regex)
+}
+
+const checkK = function (row, prop) {
```

Comment:
mis-spelling?

Reply:
No :)

Naming convention giving us an odd name.  Field in the spec is called 'QnK' where n is the question number, so it becomes checkK.

Happy?